### ISSUE TITLE
Backing up a Draft schema in Nebula studio

### ISSUE THREAD

**Issue by wey-gu** [vesoft-inc/nebula-studio#618]
### Discussed in https://github.com/vesoft-inc/nebula/discussions/5681

<div type='discussions-op-text'>

<sup>Originally posted by **PabloPete** August 17, 2023</sup>
I have a Nebula studio instance that is not running in a Docker container and have created a Draft Schema that several people are editing. Is it possible to back this up and restore it and not just the PNG image ?</div>

----

We need to keep iterating a schema for a long time, to enable that, some sort of file serialization may be needed.

**Comment by PabloPete** [comment]
We are a small team of Golang Devs. Can we help contribute to a solution ? If so what would be the starting point ?

**Comment by wey-gu** [comment]
> We are a small team of Golang Devs. Can we help contribute to a solution ? If so what would be the starting point ?

Wow, amazing! @PabloPete , thanks a lot for this, would you mind sharing proposals on how could this be done so that we could further discuss within the community?

cc @hetao92 @mizy @slimshadylol0303

**Comment by PabloPete** [comment]
If some details on the structure of what is generated by a schema draft can be shared along with where it can be extracted from then we can propose and develop an export tool that provides a version that can be saved as a standalone file that can be re-imported from disk.

**Comment by mizy** [comment]
> If some details on the structure of what is generated by a schema draft can be shared along with where it can be extracted from then we can propose and develop an export tool that provides a version that can be saved as a standalone file that can be re-imported from disk.

good idea , maybe we can export the schema draft data to a json file? and so we don't need provide server api now , just add a export json button

**Comment by wey-gu** [comment]
After we have the serlization on the schema draft, we could think big on more like wiring to datasets with the importer configuration, which, could end up as a bundle to be shared among community users, then a hub could be created to allow the sharing of the schema/(schema with datasets), etc.

**Comment by yyh0808** [comment]
Hello, @PabloPete .I am a product designer at Studio. I'm very glad to hear that you have contributed a great idea. If you can develop it, I suggest adding an import template function on the left side of the image and an export sketch template function on the right side. I have roughly marked them with red boxes. You can develop a demo and I can contribute some designs based on your demo :)
<img width="1736" alt="截屏2023-09-12 15 26 41" src="https://github.com/vesoft-inc/nebula-studio/assets/8474977/ae3707b8-d8b8-488b-a1ed-b0e297b57f0c">

**Comment by slimshadylol0303** [comment]
Hi @PabloPete, I am the product manager of NebulaGraph, thank you so much for bring up such an awesome idea ❤️! 
The serialization on the schema draft is a valid requirement for NebulaGraph Studio, it can make schema draft version management and teamwork more convenient.
I think your proposal is wonderful, we are very glad and thankful for your solution contribution. Please keep in touch and we can discuss the solution detail to make it better.😆

**Comment by PabloPete** [comment]
Hi @slimshadylol0303! Thanks for reaching out. I'm looking for guidance on the file structure. I believe it's in tasks.db within /usr/local/nebula-graph-studio/. However, tasks.db also contains queries, unless I'm mistaken. I could try a fresh install and try pattern matches to extract only the draft schema, but I'm unsure what else is needed for readability. Copying the task.db file to another machine didn't work before which suggests there's more to it. One may need to explore the studio code base for a clearer solution. Maybe someone more familiar with the code base can help.

**Comment by PabloPete** [comment]
@yyh0808 Your UX solution looks clean and simple, I assume on the right side you will provide a drop-down menu of schemas that may be exported. I don't want to complicate things with another feature others might not find useful but a permission setting around who can export the entire schema would be useful for those collaborating who wish to protect their intellectual property.

**Comment by wey-gu** [comment]
Tagged this as a `hacktoberfest` task, but I expect this single feature could be split into multiple steps that deserved more credit from `hacktoberfest` event.

**Comment by PabloPete** [comment]
Did this gain any momentum during hacktoberfest ? Is there anything to feed back yet ?

**Comment by zhanghuidinah** [comment]
Hi, @PabloPete @slimshadylol0303 Hacktoberfest for developers has started for the new year. If you're interested, you're welcome to join us.
